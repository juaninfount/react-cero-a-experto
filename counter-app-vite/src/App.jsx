//import { useState } from 'react'
//import './App.css'
import { PropTypes } from 'prop-types';

export const App = ({title, subtitle, name}) => {
  //const [count, setCount] = useState(0)
  //console.log(title);

  //if(!subtitle) throw new Error('Error: definir subtitulo');

  return (
    <>      
      <h1>{title}</h1>
      <p>{subtitle}</p>
      <p>{name}</p>      
    </>
  )
};

App.propTypes = {
  name: PropTypes.string,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
};

App.defaultProps = {
  name: 'Juan Perez',
  //title: 'No hay titulo',
  subtitle: 'No hay subtitulo'
}
//export default App
