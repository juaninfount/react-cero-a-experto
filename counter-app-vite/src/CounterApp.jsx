import { useState, React, useReducer } from "react";
import { PropTypes } from "prop-types";

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);
    const add = () => {
        setCounter(counter + 1);
    };

    const substract = () => setCounter(counter -1);
    const reset = () => setCounter(0); 
    
    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button type='button' onClick={add}>+1</button>
            <button type='button' onClick={substract}>-1</button>
            <button type='button' onClick={reset}>reset</button>

        </>
    );

}