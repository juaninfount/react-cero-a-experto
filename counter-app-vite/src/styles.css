html, body{
    background-color: #1c2b64;
    color: white ;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}