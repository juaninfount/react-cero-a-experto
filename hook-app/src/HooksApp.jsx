
export const HooksApp = () => {
    return (<h1>Hooksapp</h1>);
};