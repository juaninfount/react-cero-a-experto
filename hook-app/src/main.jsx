import React    from 'react'
import ReactDOM from 'react-dom/client'
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHooks'
// import {SimpleForm} from './02-useEffect/SimpleForm' 
// import { CustomWithCustomHook } from './01-useState/CustomWithCustomHook'
//import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { FocusScreen } from './04-useRef/FocusScreen'
import { MemoHook } from './06-Memos/MemoHook'
//import { Memorize } from './06-Memos/Memorize'
import './index.css'
//import {HooksApp} from  './HooksApp'
//import {CounterApp} from './01-useState/CounterApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <CounterApp /> */}
    {/* <CustomWithCustomHook ></CustomWithCustomHook> */}
    {/* <SimpleForm ></SimpleForm> */}
    {/* <FormWithCustomHook></FormWithCustomHook> */}
    {/* <MultipleCustomHooks /> */}
    {/* <FocusScreen></FocusScreen> */}
    <MemoHook  />
  </React.StrictMode>
)
