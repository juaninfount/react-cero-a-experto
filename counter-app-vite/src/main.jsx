import React from 'react'
import ReactDOM from 'react-dom/client'
import { App } from './App'
import { CounterApp } from './CounterApp'
import './styles.css'


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <CounterApp value={0} /> */}
    <App title="Hola, Soy Vegeta" ></App>
  </React.StrictMode>,
)