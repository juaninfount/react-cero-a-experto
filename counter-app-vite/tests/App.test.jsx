import { render } from '@testing-library/react';
import { App } from '../src/App';


describe('Pruebas en <App />', () => {

    test('debe de hacer match con el snapshot', () => {
        const title = 'Hola, Soy Goku';
        const { container } = render(<App title={title} />);

        expect(container).toMatchSnapshot();
        //console.log(container);
    });

    test('debe de mostrar titulo en un h1', () => {
        const title = 'Hola, Soy Goku';
        const { container, getByText } = render(<App title={title} />);
        expect(getByText(title)).toBeTruthy();
    });
});