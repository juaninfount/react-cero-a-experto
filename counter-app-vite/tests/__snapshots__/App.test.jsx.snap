// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Juan Perez
  </p>
</div>
`;
